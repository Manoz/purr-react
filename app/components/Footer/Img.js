import styled from 'styled-components';

import BaseImg from 'components/Img';

const Img = styled(BaseImg)`
  margin: 0;
  display: inline-block;
  width: 14px;
`;

export default Img;
