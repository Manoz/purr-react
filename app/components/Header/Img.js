import styled from 'styled-components';

import BaseImg from 'components/Img';

const Img = styled(BaseImg)`
  display: block;
  margin: 0 auto;
  max-width: 100%;
`;

export default Img;
