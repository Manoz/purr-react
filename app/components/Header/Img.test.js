import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import Img from './Img';

const src = 'http://example.com/test.jpg';
const alt = 'test';

const renderComponent = (props = {}) => shallow(<Img src={src} alt={alt} {...props} />);

describe('<Img />', () => {
  it('should match the snapshot', () => {
    const wrapper = renderer
      .create(<Img src={src} alt={alt} />)
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it('should have an src attribute', () => {
    const wrapper = renderComponent();
    expect(wrapper.prop('src')).toEqual(src);
  });

  it('should have an alt attribute', () => {
    const wrapper = renderComponent();
    expect(wrapper.prop('alt')).toEqual(alt);
  });
});
