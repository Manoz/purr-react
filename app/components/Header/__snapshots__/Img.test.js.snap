// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Img /> should match the snapshot 1`] = `
.c0 {
  display: block;
  margin: 0 auto;
  max-width: 100%;
}

<img
  alt="test"
  className="c0"
  src="http://example.com/test.jpg"
/>
`;
